packages/core/src/controllers/publisher.ts

```diff
@@ -151,17 +170,35 @@ export class Publisher extends IPublisher {
     if (isUndefined(request.params?.tag)) delete request.params?.tag;
     this.logger.debug(`Outgoing Relay Payload`);
     this.logger.trace({ type: "message", direction: "outgoing", request });
-    return this.relayer.request(request);
+    const result = await this.relayer.request(request);
+    this.relayer.events.emit(RELAYER_EVENTS.publish, params);
+    this.logger.debug(`Successfully Published Payload`);
+    return result;
   }
 
   private removeRequestFromQueue(id: string) {
     this.queue.delete(id);
   }
 
   private checkQueue() {
-    this.queue.forEach(async (params) => {
-      const { topic, message, opts } = params;
-      await this.publish(topic, message, opts);
+    this.queue.forEach(async (params, id) => {
+      const attempt = params.attempt + 1;
+      this.queue.set(id, { ...params, attempt });
+      const { topic, message, opts, attestation } = params;
+      this.logger.warn(
+        {},
```

Comment:
Why are we passing an empty object here ?

Reply:
its just for the logger, there is no error object to pass